packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/create-paypal-commerce-alternative-methods-button-strategy.ts

```diff
@@ -1,37 +1,19 @@
-import { createFormPoster } from '@bigcommerce/form-poster';
-import { createRequestSender } from '@bigcommerce/request-sender';
-import { getScriptLoader } from '@bigcommerce/script-loader';
-
 import {
     CheckoutButtonStrategyFactory,
     toResolvableModule,
 } from '@bigcommerce/checkout-sdk/payment-integration-api';
 
-import {
-    PayPalCommerceIntegrationService,
-    PayPalCommerceRequestSender,
-    PayPalCommerceScriptLoader,
-} from '../index';
+import createPayPalCommerceIntegrationService from '../create-paypal-commerce-integration-service';
 
 import PayPalCommerceAlternativeMethodsButtonStrategy from './paypal-commerce-alternative-methods-button-strategy';
 
 const createPayPalCommerceAlternativeMethodsButtonStrategy: CheckoutButtonStrategyFactory<
     PayPalCommerceAlternativeMethodsButtonStrategy
-> = (paymentIntegrationService) => {
-    const { getHost } = paymentIntegrationService.getState();
-
-    const paypalCommerceIntegrationService = new PayPalCommerceIntegrationService(
-        createFormPoster(),
-        paymentIntegrationService,
-        new PayPalCommerceRequestSender(createRequestSender({ host: getHost() })),
-        new PayPalCommerceScriptLoader(getScriptLoader()),
-    );
-
-    return new PayPalCommerceAlternativeMethodsButtonStrategy(
+> = (paymentIntegrationService) =>
+    new PayPalCommerceAlternativeMethodsButtonStrategy(
         paymentIntegrationService,
```

Comment:
As i see we have paymentIntegrationService as first argument in every provider. Maybe we can move it to the same mutual file and return it from there with  `new PayPalCommerceIntegrationService`. Is it makes sense?)

Reply:
I don't think we should, because `paymentIntegrationService` is a prop that comes from another place. It could be better to keep strategy initialisation as simple as possible.